app/main.py

```diff
@@ -1 +1,37 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self._hidden: bool = False
+        Animal.alive.append(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self._hidden
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, Health: {self.health}, "
+                f"Hidden: {self._hidden}}}")
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def __sub__(self, damage: int) -> None:
+        self.health -= damage
+        if self.health <= 0:
+            self.die()
+
+    def hide(self) -> None:
+        self._hidden = not self._hidden
```

Comment:
Code Efficiency: You can change the `bool` value in one line. The `not` operator can be used to flip the boolean value of `self._hidden`. So, instead of using an `if` statement, you can simply use `self._hidden = not self._hidden`. This makes your code more concise and easier to read.